src/asset/MountArchive.cpp

```diff
@@ -0,0 +1,89 @@
+#include "MountArchive.h"
+
+#include "bit7z/bitarchivereader.hpp"
+#include "utils/Log.h"
+
+#include <cstddef>
+#include <filesystem>
+#include <format>
+#include <string>
+#include <vector>
+
+MountArchive::MountArchive(std::filesystem::path archive_path)
+    : Mount(archive_path), reader(new bit7z::BitArchiveReader(library, path.string())) {
+}
+
+MountArchive::~MountArchive() {
+    MountArchive::save_cache();
+}
+
+Mount::MountType MountArchive::get_type() {
+    return ARCHIVE;
+}
+
+bool MountArchive::load() {
+    reset_reader();
+
+    try {
+        reader = new bit7z::BitArchiveReader(library, get_path().string());
+        reader->test();
+    }
+    catch (const std::exception& exception) {
+        return false;
+    }
+
+    load_cache();
+
+    return true;
+}
+
+bool MountArchive::contains_file(std::string path) {
+    if (static_cache.contains(path)) {
+        return true;
+    }
+    return false;
+}
+
+std::vector<std::byte> MountArchive::fetch_data(std::string path) {

```

Comment:
we are using uint8_t rather than std::byte elsewhere in the project. not saying that we shouldn't use std::byte here, but we should definitely make it consistent across the codebase either way